autoload/asyncomplete/sources/ale.vim

```diff
@@ -0,0 +1,25 @@
+function! asyncomplete#sources#ale#get_source_options(...) abort
+    return extend(extend({
+    \     'name': 'ale',
+    \     'completor': function('asyncomplete#sources#ale#completor'),
+    \     'whitelist': ['*'],
+    \     'triggers': asyncomplete#sources#ale#get_triggers(),
+    \ }, a:0 >= 1 ? a:1 : {}), {'refresh_pattern': '\k\+$'})
```

Comment:
Split this into a variable with the first `extend` call, and `return extend(l:that_variable, {'refresh_pattern': '\k\+$'})`. That'll be easier to follow.